homeassistant/components/schlage/coordinator.py

```diff
@@ -0,0 +1,39 @@
+"""DataUpdateCoordinator for the Schlage integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pyschlage import Lock, Schlage
+from pyschlage.exceptions import Error
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER, UPDATE_INTERVAL
+
+
+@dataclass
+class SchlageData:
+    """Container for cached data from the Schlage API."""
+
+    locks: dict[str, Lock]
+
+
+class SchlageDataUpdateCoordinator(DataUpdateCoordinator[SchlageData]):
+    """The Schlage data update coordinator."""
+
+    def __init__(self, hass: HomeAssistant, api: Schlage) -> None:
+        """Initialize the class."""
+        super().__init__(hass, LOGGER, name=DOMAIN, update_interval=UPDATE_INTERVAL)
```

Comment:
Ideally we use a unique name here instead of the DOMAIN since they may have more than one

Reply:
Maybe use the domain + config entry title

Reply:
Set it to `f"{DOMIAN} ({username})"`. (The config entry title is the username, but referencing username here seems clearer)